clang/tools/driver/cc1depscan_main.cpp

```diff
@@ -515,8 +529,14 @@ int cc1depscan_main(ArrayRef<const char *> Argv, const char *Argv0,
             Args.getLastArg(clang::driver::options::OPT_dump_depscan_tree_EQ))
       DumpDepscanTree = Saver.save(Arg->getValue());
 
-    CAS = reportAsFatalIfError(
-        llvm::cas::createOnDiskCAS(llvm::cas::getDefaultOnDiskCASPath()));
+    // FIXME: Sniff CASOptions from CC1Args->getValues() instead of using
```

Comment:
For this one, I think we can teach Driver to pass all the correct CAS options and pass them before the `-cc1-args` args (if `-fcas-path` is not a cc1 only option).

Reply:
That's true.

Is there a specific reason you'd prefer outside of `-cc1-args`?

My intuition is that it's better to sniff from `-cc1-args`. I don't like being able to specify `-fcas-path` in both places, since the earlier one in the command-line would "win" and clobber the other... even if it was empty. I would find the "specify in either place" confusing, since I might specify in the wrong place and it would get clobbered.